src/Spryker/Client/RabbitMq/RabbitMqConfig.php

```diff
@@ -75,34 +106,89 @@ protected function getQueueConnectionConfigs(): array
     {
-        $queueOptionCollection = new ArrayObject();
-        $queueOptionCollection->append(new RabbitMqOptionTransfer());
+        if ($this->queueOptionCollection !== null) {
+            return $this->queueOptionCollection;
+        }
 
-        return $queueOptionCollection;
+        $queueConfigurations = $this->getQueueConfiguration();
+        $this->queueOptionCollection = new ArrayObject();
+
+        foreach ($queueConfigurations as $queueNameKey => $queueConfiguration) {
+            if (!is_array($queueConfiguration)) {
+                $this->queueOptionCollection->append($this->createQueueOption($queueConfiguration, sprintf('%s.error', $queueConfiguration), 'error'));
+
+                continue;
+            }
+
+            foreach ($queueConfiguration as $routingKey => $queueName) {
+                $this->queueOptionCollection->append($this->createQueueOption($queueNameKey, $queueName, $routingKey));
+            }
+        }
+
+        return $this->queueOptionCollection;
     }
 
     /**
      * @return array
      */
-    public function getMessageConfig(): array
+    protected function getQueueConfiguration(): array
     {
-        return [
-            'delivery_mode' => AMQPMessage::DELIVERY_MODE_PERSISTENT,
-        ];
+        return [];
     }
 
     /**
-     * @return \Generated\Shared\Transfer\QueueConnectionTransfer
+     * @param string $queueName
+     * @param string $boundQueueName
+     * @param string $routingKey
+     *
+     * @return \Generated\Shared\Transfer\RabbitMqOptionTransfer
      */
-    public function getDefaultQueueConnectionConfig(): QueueConnectionTransfer
+    protected function createQueueOption($queueName, $boundQueueName, $routingKey)
     {
-        return (new QueueConnectionTransfer())
-            ->setInsist(static::AMQP_STREAM_CONNECTION_INSIST)
-            ->setLoginMethod(static::AMQP_STREAM_CONNECTION_LOGIN_METHOD)
-            ->setConnectionTimeout(static::AMQP_STREAM_CONNECTION_CONNECTION_TIMEOUT)
-            ->setReadWriteTimeout(static::AMQP_STREAM_CONNECTION_READ_WRITE_TIMEOUT)
-            ->setKeepAlive(static::AMQP_STREAM_CONNECTION_KEEP_ALIVE)
-            ->setHeartBeat(static::AMQP_STREAM_CONNECTION_HEART_BEAT)
-            ->setChannelRpcTimeout(static::AMQP_STREAM_CONNECTION_CHANNEL_RPC_TIMEOUT);
+        $queueOptionTransfer = new RabbitMqOptionTransfer();
+        $queueOptionTransfer
+            ->setQueueName($queueName)
+            ->setDurable(true)
+            ->setType('direct')
+            ->setDeclarationType(Connection::RABBIT_MQ_EXCHANGE)
+            ->addBindingQueueItem($this->createQueueBinding($queueName))
+            ->addBindingQueueItem($this->createErrorQueueBinding($boundQueueName, $routingKey));
+
+        return $queueOptionTransfer;
+    }
+
+    /**
+     * @param string $queueName
+     *
+     * @return \Generated\Shared\Transfer\RabbitMqOptionTransfer
+     */
+    protected function createQueueBinding($queueName)
```

Comment:
this method is exactly like the `createErrorQueueBinding` but just with an empty `routingKey` maybe you can just merge these two methods together and please drop `error` from any naming in core